pkg/products/threescale/reconciler.go

```diff
@@ -342,6 +383,260 @@ func (r *Reconciler) backupSystemSecrets(ctx context.Context, serverClient k8scl
+			Type: envoyapi.Cluster_STRICT_DNS,
+		},
+		ConnectTimeout: &duration.Duration{
+			Seconds: 1,
+		},
+		LbPolicy:             envoyapi.Cluster_ROUND_ROBIN,
+		Http2ProtocolOptions: &envoycore.Http2ProtocolOptions{},
+
+		LoadAssignment: &envoyapi.ClusterLoadAssignment{
+			ClusterName: "ratelimit",
+			NamedEndpoints: map[string]*endpoint.Endpoint{
+				"socket-address": {
+					Address: &envoycore.Address{
+						Address: &envoycore.Address_SocketAddress{
+							SocketAddress: &envoycore.SocketAddress{
+								Address: "10.130.2.40",
+								PortSpecifier: &envoycore.SocketAddress_PortValue{
+									PortValue: 8081,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	/*	var domainsArray []string
+		domainsArray[0] = "*"
+
+		connectionManager := hcm.HttpConnectionManager{
+			RouteSpecifier: &hcm.HttpConnectionManager_RouteConfig{
+				RouteConfig: &envoyapi.RouteConfiguration{
+					Name: "local-route",
+					VirtualHosts: []*route.VirtualHost{{
+						Name: dcName,
+						Domains: domainsArray,
+						Routes: []*route.Route{{
+							Match: &route.RouteMatch{
+								PathSpecifier: &route.RouteMatch_Prefix{
+									Prefix: "/",
+								},
+							},
+						},
+						},
+					},
+
+					},
+				},
+			},
+		}*/
+
+	listener := envoyapi.Listener{
+		Name: "http",
+		Address: &envoycore.Address{
+			Address: &envoycore.Address_SocketAddress{
+				SocketAddress: &envoycore.SocketAddress{
+					Protocol: envoycore.SocketAddress_TCP,
+					Address:  "0.0.0.0",
+					PortSpecifier: &envoycore.SocketAddress_PortValue{
+						PortValue: uint32(8443),
+					},
+				},
+			},
+		},
+		FilterChains: []*listener.FilterChain{{
+			Filters: []*listener.Filter{{
+				Name:       "envoy.http_connection_manager",
+				ConfigType: &listener.Filter_Config{},
+			}},
+		}},
+	}
+
+	clusterJsonMarshalled, err := json.Marshal(cluster)
+	if err != nil {
+		return fmt.Errorf("failed to create the %s envoy config: %w", deploymentConfig.Name, err)
+
+	}
+	ratelimitClusterJsonMarshalled, err := json.Marshal(rateLimitCluster)
+	if err != nil {
+		return fmt.Errorf("failed to create the %s envoy config: %w", deploymentConfig.Name, err)
+
+	}
+	listenerJsonMarshalled, err := json.Marshal(listener)
+	if err != nil {
+		return fmt.Errorf("failed to create the %s envoy config: %w", deploymentConfig.Name, err)
+
+	}
+
+	clusterYamlMarshalled, err := k8syaml.JSONToYAML(clusterJsonMarshalled)
+	if err != nil {
+		return fmt.Errorf("failed to create the %s envoy config: %w", deploymentConfig.Name, err)
+
+	}
+	rateLimitClusterYamlMarshalled, err := k8syaml.JSONToYAML(ratelimitClusterJsonMarshalled)
+	if err != nil {
+		return fmt.Errorf("failed to create the %s envoy config: %w", deploymentConfig.Name, err)
+
+	}
+	listenerMarshalledYaml, err := k8syaml.JSONToYAML(listenerJsonMarshalled)
+	if err != nil {
+		return fmt.Errorf("failed to create the %s envoy config: %w", deploymentConfig.Name, err)
+
+	}
```

Comment:
Maybe we could create a utility function `JSONObjectToYAML` that'd look like this

```go
func JSONObjectToYAML(o interface{}) ([]byte, error) {
  jsonMarshelled, err := json.Marshal(o)
  if err := nil {
    return nil, err
  }

  return k8syaml.JSONToYAML(jsonMarshalled)
}
```

It'd make this section shorter and more readable